components/src/Bloomreach/bloomreach.appheadersearch.main.tsx

```diff
@@ -0,0 +1,180 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React, { FormEvent, SyntheticEvent } from 'react';
+import { bloomreachSuggestionSearch } from '../utils/BloomreachSearchService';
+import './bloomreach.headersearch.main.less';
+
+const intl = { get: str => str };
+
+interface BloomreachHeaderSearchMainProps {
+    isMobileView: boolean,
+    onSearchPage?: (...args: any[]) => any,
+}
+interface BloomreachHeaderSearchMainState {
+    keywords: string,
+    suggestions: [],
+}
+
+class BloomreachHeaderSearchMain extends React.Component<BloomreachHeaderSearchMainProps, BloomreachHeaderSearchMainState> {
+  private searchInput: React.RefObject<HTMLInputElement>;
+
+  private suggestionListElements: HTMLLIElement[];
+
+  private suggestionIndex: number;
+
+  static defaultProps = {
+    onSearchPage: () => {},
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      keywords: '',
+      suggestions: [],
+    };
+    this.searchInput = React.createRef();
+    this.suggestionListElements = [];
+    this.suggestionIndex = 0;
+    this.setSuggestionsList = this.setSuggestionsList.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.search = this.search.bind(this);
+    this.inputHandleKeyDown = this.inputHandleKeyDown.bind(this);
+    this.liHandleKeyDown = this.liHandleKeyDown.bind(this);
+  }
+
+  setSuggestionsList(element, indexOfSuggestion: number) {
+    this.suggestionListElements[indexOfSuggestion] = element;
+  }
+
+  handleChange(event) {
+    const keywords = event.target.value;
+
+    bloomreachSuggestionSearch(keywords)
+      .then((res) => {
+        const brSuggestions = res.response.suggestions;
+        const suggestions = brSuggestions ? brSuggestions.map(suggestion => suggestion.dq) : [];
+
+        this.setState({
+          keywords,
+          suggestions,
+        });
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error(error.message);
+      });
+  }
+
+  search(event?: SyntheticEvent, listElementKeyword?: string) {
+    const { onSearchPage } = this.props;
+    const { keywords } = this.state;
+
+    if (listElementKeyword) {
+      document.querySelector('.collapsable-container').classList.remove('show');
+      onSearchPage(listElementKeyword);
+    } else if (keywords !== '') {
+      document.querySelector('.collapsable-container').classList.remove('show');
+      onSearchPage(keywords);
+    }
+
+    this.searchInput.current.value = '';
+    if (event != null) {
+      event.preventDefault();
+    }
+  }
+
+  liHandleKeyDown(e) {
+    if (e.keyCode === 38) {
```

Comment:
Do you still have to use keycodes for this? How did this work on mobile?

Reply:
Adding key codes was needed to support arrow functionality with the suggestion dropdown on desktop.  Shouldn't affect mobile? Although I did just try it on my Iphone xs max and there are some issues with clicking the dropdown.  I think its an unrelated to keycodes though.

Reply:
Yeah can confirm it has nothing to do with keycodes.  Was missing touch events needed for mobile interaction.

Reply:
Fixed the mobile scrolling interaction here #416